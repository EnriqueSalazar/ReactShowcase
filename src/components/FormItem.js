/**
 * Created by enriq on 17/11/16.
 */
import React from "react";

import {ControlLabel, FormControl} from "react-bootstrap";

const FormItem = props => {
        const { callback, label, list, value, } = props;
        return (
            <div>
                <ControlLabel>{label}</ControlLabel>
                <FormControl
                    componentClass="select"
                    onChange={callback}
                    value={value}
                >
                    <option value="">{"Select"}</option>
                    {list && list.map((item, i) => {
                        return (
                            <option value={item} key={i}>
                                {item}
                            </option>
                        );
                    })}
                </FormControl>
            </div>
        );
};

export default FormItem;
