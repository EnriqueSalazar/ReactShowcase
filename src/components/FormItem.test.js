/**
 * Created by enriq on 17/11/16.
 */
import React from "react";
import FormItem from ".//FormItem";
import renderer from "react-test-renderer";

test("Render FormItem test", () => {
  let testCallback = () => {};
  const component = renderer.create(
    (
      <FormItem
        callback={testCallback}
        label="Brands"
        list={[ { value: 1, option: "uno" }, { value: 2, option: "dos" } ]}
      />
    )
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
