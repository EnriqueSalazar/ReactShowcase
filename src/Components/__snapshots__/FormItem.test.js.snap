exports[`test Render FormItem test 1`] = `
<div
  className="form-group">
  <label
    className="control-label"
    htmlFor="formControlsSelect">
    Brands
  </label>
  <select
    className="form-control"
    id="formControlsSelect"
    onChange={[Function]}
    type={undefined}>
    <option
      value="select">
      
    </option>
    <option
      value={1}>
      uno
    </option>
    <option
      value={2}>
      dos
    </option>
  </select>
</div>
`;
